src/common/utils/time/time.ts

```diff
@@ -222,3 +219,16 @@ export function formatTimeBasedOnGranularity(range: TimeRange, granularity: Dura
 export function formatGranularity(granularity: string): string {
   return granularity.replace(/^PT?/, '');
 }
+
+export function isFullyDefinedDate(date: string): boolean {
```

Comment:
I would probably change *is* to *maybe* as the function tests necessity rather than sufficiency